balancer/frontend-v2
# Hotfix: Fix price impact calculations

# Description

Our price impact calculation function is currently quite broken (it was made years ago and never had tests and no one really knew how it worked). It previously only worked if the sell token was 18 decimals AND the swap type was SwapExactIn.

This PR adds many tests and documentation for this function as well as makes it work with all swap types and tokens.

## Type of change


## How should this be tested?

- Ensure that price impact looks correct for large swaps of USDC/USDT -> other tokens, or SwapExactOut swaps of any tokens. 
- Ensure that trading from USDT -> Y2K on Arbitrum shows a high price impact.

## Visual context

![price-impact-usdc](https://user-images.githubusercontent.com/525534/228766395-b8f1d2c8-a522-455c-a0eb-aff61eeb0f4f.png)
![price-impact-new](https://user-images.githubusercontent.com/525534/228766425-e8464fb5-60aa-4304-be7c-f52c452d08e8.png)

## Checklist:
